libs/fluxc-plugin/src/main/kotlin/org/wordpress/android/fluxc/persistence/dao/ProductsDao.kt

```diff
@@ -0,0 +1,114 @@
+package org.wordpress.android.fluxc.persistence.dao
+
+import androidx.room.Dao
+import androidx.room.Query
+import androidx.room.Upsert
+import kotlinx.coroutines.flow.Flow
+import org.wordpress.android.fluxc.model.WCProductModel
+
+@Dao
+internal abstract class ProductsDao {
+
+    @Query(
+        """
+            SELECT * FROM ProductEntity
+            WHERE localSiteId = :localSiteId
+            AND (:status IS NULL OR status = :status)
+            AND (:stockStatus IS NULL OR stockStatus = :stockStatus)
+            AND (:type IS NULL OR type = :type)
+            AND (:category IS NULL OR categories LIKE '%' || :category || '%')
+            AND (:excludeSampleProducts IS NULL OR isSampleProduct = :excludeSampleProducts)
```

Comment:
I don't think this condition is correct, because `excludeSampleProducts` is not nullable, and `isSampleProduct` is not equivalent to `excludeSampleProducts`.

I think something like the following would work, WDYT?
```suggestion
            AND (:excludeSampleProducts = 0 OR isSampleProduct = 0)
```

Reply:
Very good point, thanks. Addressed in a131423c1fc0f11ffcb7a441474d2647cc45682c